lib/widgets/user_profile.dart

```diff
@@ -5,283 +5,382 @@ import 'package:intl/intl.dart';
 
-  // TODO: add `.fromUser` constructor
   UserProfile({@required int userId, @required this.instanceUrl})
-      : _userView = LemmyApi(instanceUrl)
-            .v1
-            .getUserDetails(
-                userId: userId, savedOnly: true, sort: SortType.active)
-            .then((res) => res.user);
+      : _userDetails = LemmyApi(instanceUrl).v1.getUserDetails(
+            userId: userId, savedOnly: false, sort: SortType.active);
 
-  UserProfile.fromUserView(UserView userView)
-      : _userView = Future.value(userView),
-        instanceUrl = userView.instanceUrl;
+  UserProfile.fromUserDetails(UserDetails userDetails)
+      : _userDetails = Future.value(userDetails),
+        instanceUrl = userDetails.user.instanceUrl;
 
   @override
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
-    final colorOnTopOfAccentColor =
-        textColorBasedOnBackground(theme.accentColor);
+    final userDetailsSnap = useFuture(_userDetails, preserveState: false);
 
-    final userViewSnap = useFuture(_userView, preserveState: false);
+    if (!userDetailsSnap.hasData) {
+      return const Center(child: CircularProgressIndicator());
+    }
 
-    final bio = () {
-      if (userViewSnap.hasData) {
-        if (userViewSnap.data.bio != null) {
-          return Padding(
-            padding: const EdgeInsets.all(10),
-            child: Text(userViewSnap.data.bio),
-          );
-        } else {
-          return Center(
-            child: Text(
-              'no bio',
-              style: const TextStyle(fontStyle: FontStyle.italic),
-            ),
-          );
-        }
-      } else {
-        return Center(child: CircularProgressIndicator());
-      }
-    }();
+    final userView = userDetailsSnap.data.user;
 
-    Widget tabs() => DefaultTabController(
-          length: 3,
-          child: Column(
-            children: [
-              TabBar(
-                labelColor: theme.textTheme.bodyText1.color,
-                tabs: [
-                  Tab(text: 'Posts'),
-                  Tab(text: 'Comments'),
-                  Tab(text: 'About'),
-                ],
-              ),
-              Expanded(
-                child: TabBarView(
-                  children: [
-                    Center(
-                      child: Text(
-                        'Posts',
-                        style: const TextStyle(fontSize: 36),
-                      ),
-                    ),
-                    Center(
-                      child: Text(
-                        'Comments',
-                        style: const TextStyle(fontSize: 36),
-                      ),
-                    ),
-                    bio,
-                  ],
-                ),
-              )
-            ],
+    return DefaultTabController(
+      length: 3,
+      child: NestedScrollView(
+        headerSliverBuilder: (_, __) => [
+          SliverAppBar(
+            pinned: true,
+            expandedHeight: 255,
```

Comment:
increase expanded height cuz cake day date is too close to tab titles
![image](https://user-images.githubusercontent.com/10037914/94927846-1e64fa00-04c3-11eb-9816-382bd96f3fe1.png)
the space should be about the same as the one between user badges and "joined ..." text